tmt/checks/__init__.py

```diff
@@ -100,6 +101,11 @@ class Check(
         default=True,
         is_flag=True,
         help='Whether the check is enabled or not.')
+    result: Any = field(
+        default='respect',
+        serialize=lambda result: result.value if isinstance(result, enum.Enum) else str(result),
+        unserialize=lambda spec: spec,
+        help='How to interpret the check result.')
```

Comment:
I think we can switch from `Any` to `ResultInterpret`, if we add the `from_spec()` method to it, `unserialize` could use it. Might need also `normalize` callback, but I'd pay that price for getting rid of `Any`. With `Any`, no user of this field can't be sure what the value is.